openedx/core/djangoapps/user_api/accounts/tests/test_api.py

```diff
@@ -312,6 +319,35 @@ def test_create_account(self):
 
 
 @attr(shard=2)
+@pytest.mark.django_db
+def test_create_account_duplicate_email(django_db_use_migrations):
+    """
+    Test case for duplicate email constraint
+    Email uniqueness constraints were introduced in a database migration,
+    which we disable in the unit tests to improve the speed of the test suite
+
+    This test only runs if migrations have been run.
+
+    django_db_use_migrations is a pytest_django fixture which tells us whether
+    migrations are being used. Since pytest fixtures don't play nice with
+    TestCase objects this is a function and doesn't get to use assertRaises.
+    """
+    password = 'legit'
+    email = 'zappadappadoo@example.com'
+
+    if django_db_use_migrations:
+        create_account('zappadappadoo', password, email)
+        success = False
+        try:
+            create_account('different_user', password, email)
+            success = True
+        except AccountUserAlreadyExists:
+            pass
```

Comment:
There's a [pytest.raises](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions) context manager that works almost exactly like the `self.assertRaises()` method you're replacing here.